--- a/gfx/drivers/vita2d_gfx.c
+++ b/gfx/drivers/vita2d_gfx.c
@@ -217,7 +217,7 @@ static void gfx_display_vita2d_draw(void *data, unsigned video_width, unsigned v
       vertices[i].a = *color++;
    }
 
-   vita2d_draw_array_textured_mat(texture, vertices, draw->coords->vertices, &vita2d->mvp_no_rot);
+   vita2d_draw_array_textured_mat(texture, vertices, draw->coords->vertices, (float *)vita2d->mvp_no_rot.data);
 }
 
 static void gfx_display_vita2d_scissor_begin(void *data,
